File: app/src/main/java/com/fractalwrench/crazycats/mocks/MockImageDetailView.java
package com.fractalwrench.crazycats.mocks;

import android.support.annotation.NonNull;

import com.fractalwrench.crazycats.image.data.ImageData;
import com.fractalwrench.crazycats.image.detail.ImageDetailView;


public class MockImageDetailView implements ImageDetailView {

    public String errorMessage;
    public ImageData imageData;

    @Override
    public void showProgress() {

    }

    @Override
    public void showContent(@NonNull ImageData ImageData) {
        this.imageData = ImageData;
    }

    @Override
    public void showError(String errorMessage) {
        this.errorMessage = errorMessage;
    }
}

File: app/src/main/java/com/fractalwrench/crazycats/mocks/MockImageDetailView.kt
package com.fractalwrench.crazycats.mocks

import com.fractalwrench.crazycats.image.data.ImageData
import com.fractalwrench.crazycats.image.detail.ImageDetailView


class MockImageDetailView : ImageDetailView {

    var errorMessage: String? = null
    var imageData: ImageData? = null

    override fun showProgress() {

    }

    override fun showContent(content: ImageData) {
        this.imageData = content
    }

    override fun showError(errorMessage: String) {
        this.errorMessage = errorMessage
    }
}
